front-end/src/app/shared/components/contact-form/contact-form.component.ts

```diff
@@ -107,6 +107,8 @@ export class ContactFormComponent extends DestroyerComponent implements OnInit {
           this.candidateDistrictOptions = [];
         }
       });
+    this.form?.get('candidate_id')?.addAsyncValidators(this.contactService.fecIdValidator);
```

Comment:
I wonder if we should add debounce to these as you mentioned so we're not hitting the api too much:

https://stackoverflow.com/questions/49090659/angular-reactive-forms-debounce-only-some-specific-form-control#answer-49093827

Alternatively, you could add it to the validator fn below like here https://stackoverflow.com/questions/36919011/how-to-add-debounce-time-to-an-async-validator-in-angular-2#answer-62662296

Reply:
While testing, it looks like the synchronous pattern validation rule is throwing an error while the id string is less than 9 characters long and blocking the async validation rule from firing. I am only seeing network activity to check the uniqueness of the id when the string reaches 9 characters long.

![image](https://github.com/fecgov/fecfile-web-app/assets/2971785/7e8d3282-6ccd-45fa-b82d-80be459015b8)

So at the moment it doesn't look like debouncing is needed but I think it would be nice to put in anyway because at some point we may be asked to clean up that pattern error message and that may remove the sync validation block.

Reply:
I was in fact relying on the pattern validator to limit the usage of the async validator. the uniqueness validator is only called when the id is a complete and valid id, so it's infrequent.  But I went ahead and added the debounce to Matt's point that the pattern validator could change.